src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -79,9 +80,55 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.ubuntu_pro_client_all_updates_cached = []
         self.ubuntu_pro_client_all_updates_versions_cached = []
 
-    def refresh_repo(self):
-        self.composite_logger.log("\nRefreshing local repo...")
-        self.invoke_package_manager(self.repo_refresh)
+    # region Sources Management
+    def __get_custom_sources_to_spec(self, max_patch_published_date=str(), base_classification=str()):
+        # type: (str, str) -> str
+        """ Prepares the custom sources list for use in a command. Idempotent. """
+        try:
+            formula = "F-[{0}]-[{1}]".format(max_patch_published_date, base_classification)
+            if self.cached_customer_source_list_formula == formula:
+                return self.custom_sources_list
+            self.cached_customer_source_list_formula = formula
+
+            command = self.cmd_prep_custom_sources_list_template.replace("<CLASSIFICATION>", base_classification if base_classification == "security" else "\"\"")
+            code, out = self.env_layer.run_command_output(command, False, False)
+            sources_content = self.env_layer.file_system.read_with_retry(self.custom_sources_list)
+            self.composite_logger.log_debug("[APM] Modified custom sources list with classification. [Code={0}][Out={1}][Content={2}]".format(str(code), str(out), str(sources_content)))   # non-zero error code to be investigated
```

Comment:
maybe nit: Is this modifying sources content or simply fetching all Security updates?

Reply:
94 changes sources in the file we use.
95 reads it back to us for logging. 96 logs it.